pkg/providers/v1/retry_handler.go

```diff
@@ -173,3 +181,102 @@ func (b *Backoff) ReportError() {
 
 	b.countErrorsRequestLimit += 1.0
 }
+
+// Standard retry implementation, except that it doesn't retry RequestLimitExceeded errors.
+// This works in tandem with CrossRequestRetryDelay, which reports these errors before
+// the middleware exits upon seeing this error
+type customRetryer struct {
+	awsv2.Retryer
+}
+func (r customRetryer) IsErrorRetryable(err error) bool {
+	var ae smithy.APIError
+	if errors.As(err, &ae) && strings.Contains(ae.Error(), NON_RETRYABLE_ERROR) {
+		return false
+	}
+	return r.Retryer.IsErrorRetryable(err)
+}
```

Comment:
Why do we need this? I didn't see it used anywhere. 

v2 retryer itself has an implementation to decide which errors are retryable https://github.com/aws/aws-sdk-go-v2/blob/main/aws/retry/retryable_error.go#L15. But I didn't see the purpose of custom retryer here